cmd/dinghy.go

```diff
@@ -133,46 +152,78 @@ func dinghyRender(args []string) (string, error) {
+		}
 
-			errValidation := builder.ValidatePipelines(d, out.Bytes())
-			if errValidation != nil {
-				log.Errorf("Final Dinghyfile failed validations, please correct them and retry. %v", errValidation)
-				return "", fmt.Errorf("Final Dinghyfile failed validations, please correct them and retry. %v", errValidation)
-			}
+		log.Info("Parsing final dinghyfile to struct for validation")
+		d, err := dinghyfileStruct(builder, out)
+		if err != nil {
+			log.Errorf("Parsing to struct failed: %v", err)
+			return "", fmt.Errorf("Parsing to struct failed: %v", err)
+		}
 
-			errValidation = builder.ValidateAppNotifications(d, out.Bytes())
-			if errValidation != nil {
-				log.Errorf("Final Dinghyfile failed application notification validations, please correct them and retry. %v", errValidation)
-				return "", fmt.Errorf("Final Dinghyfile failed application notification validations, please correct them and retry. %v", errValidation)
-			}
+		errValidation := builder.ValidatePipelines(d, out.Bytes())
+		if errValidation != nil {
+			log.Errorf("Final Dinghyfile failed validations, please correct them and retry. %v", errValidation)
+			return "", fmt.Errorf("Final Dinghyfile failed validations, please correct them and retry. %v", errValidation)
+		}
 
-			//Save file if output exists
-			//Log output
-			var outIndent bytes.Buffer
-			json.Indent(&outIndent, out.Bytes(), "", "  ")
-			saveOutputFile(viper.GetString("output"), outIndent)
-			log.Info("Output:\n")
-			fmt.Println(outIndent.String())
-			log.Info("Final dinghyfile is a valid JSON Object.")
+		errValidation = builder.ValidateAppNotifications(d, out.Bytes())
+		if errValidation != nil {
+			log.Errorf("Final Dinghyfile failed application notification validations, please correct them and retry. %v", errValidation)
+			return "", fmt.Errorf("Final Dinghyfile failed application notification validations, please correct them and retry. %v", errValidation)
+		}
+
+		//Save file if output exists
+		//Log output
+		outIndent := indentType(docType, out.Bytes())
+		buff:=bytes.NewBuffer(outIndent)
+		saveOutputFile(viper.GetString("output"), buff)
+		log.Info("Output:\n")
+		fmt.Println(buff.String())
+		log.Info(fmt.Sprintf("Final dinghyfile is a valid %v Object.", strings.ToUpper(docType)))
+
+		return buff.String(), nil
+
+	}
+	return "", nil
+}
+
+func indentType(docType string, i []byte) []byte {
+	switch  docType {
+	case "json":
+		var outIndent bytes.Buffer
+		json.Indent(&outIndent, i, "", "  ")
+		return outIndent.Bytes()
+	}
+	return i
+}
 
-			return outIndent.String(), nil
+func validateType(docType string, i []byte) bool {
+	switch  docType {
+	case "json":
+		return json.Valid(i)
+	case "yaml":
+		var validate map[string]interface{}
+		erryaml := yaml.Unmarshal(i, &validate)
+		if erryaml != nil {
+			fmt.Errorf("validation for %v failed: %v", docType, erryaml)
+			return false
 		}
+		return true
+	default:
+		fmt.Errorf("Invalid docType: %v", docType)
```

Comment:
These errors don't look like they're being consumed, should we updated the fn to return `(bool, error)` instead and print the error?